### ISSUE TITLE
[feature request] encourage the usage of named import

### ISSUE THREAD

**Issue by xboy2012** [yannickcr/eslint-plugin-react#3100]
As the doc says
https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports

named import is preferred over the default import

Good:

```tsx
import { useState } from 'react';

// useState(...)
```

Bad: 

```tsx
import React from 'react';

// React.useState(...)
```

There should be a rule to enforce that?

**Comment by rygine** [comment]
For a solution to this, see https://github.com/yannickcr/eslint-plugin-react/issues/2628#issuecomment-984160944.

**Comment by MiroslavPetrik** [comment]
@ljharb How is this duplicate?

`plugin:react/jsx-runtime` will mark the `import React from 'react'` as error, when you have plain JSX used, so there is the 'hidden' `React.createElement` for which the react was needed, but not anymore.

But the OP has code `React.useState` for which the `React` import is needed, and the rule should error on the code, as the direct hook import is preferred. 

I didn't find better rule than this: https://github.com/jsx-eslint/eslint-plugin-react/issues/2628#issuecomment-984160944

**Comment by ljharb** [comment]
ahh, i see what you mean. you want a specific rule to force React's adjacent imports - Component, PureComponent, hooks, etc - to be either named-imported (or CJS-destructured) vs being chained off of React/the default import?

This is related to #2628; specifically https://github.com/jsx-eslint/eslint-plugin-react/issues/2628#issuecomment-736680314. https://github.com/jsx-eslint/eslint-plugin-react/issues/2628#issuecomment-984160944 provides a workaround in the meantime - but i think i'm going to stick with my stated opinion:

> when ["react stops providing a default export"] happens, you won’t need a linter rule to enforce [named imports] - your [default] imports just won’t work.

**Comment by MiroslavPetrik** [comment]
The default export is half the story, still you can do wildcard import `import * as React from 'react'`
so the rule to not have global `React` is still needed.

The workaround does not cover this case.

**Comment by ljharb** [comment]
@MiroslavPetrik for that, `import *` should just be banned wholesale, using no-restricted-syntax. I'd also be willing to entertain a rule for that in eslint-plugin-import - but it's not unique to React.

**Comment by aaronadamsCA** [comment]
Also note this is very easy to do with `no-restricted-imports`:

```js
{
  name: "react",
  importNames: ["default"],
  message: "Use named imports instead",
}
```

This catch both `import React` and `import * as React` syntax.

**Comment by tkrotoff** [comment]
> ```js
> {
>   name: "react",
>   importNames: ["default"],
>   message: "Use named imports instead",
> }
> ```
> 
> This catch both `import React` and `import * as React` syntax.

THIS should be done by default by eslint-plugin-react

**Comment by man-trackunit** [comment]
> Also note this is very easy to do using [`no-restricted-imports`](https://eslint.org/docs/latest/rules/no-restricted-imports):
> 
> "no-restricted-imports": ["error", {
>   "paths": [{
>     "name": "react",
>     "importNames": ["default"],
>     "message": "Please use named imports instead."
>   }]
> }]
> This should catch both `import React` and `import * as React` imports.

Yes, but would be lovely with an auto-fix 🤘
